synapse/federation/sender/per_destination_queue.py

```diff
@@ -347,32 +343,33 @@ def _pop_pending_edus(self, limit):
 
     @defer.inlineCallbacks
     def _get_new_device_messages(self):
-        last_device_stream_id = self._last_device_stream_id
-        to_device_stream_id = self._store.get_to_device_stream_token()
-        contents, stream_id = yield self._store.get_new_device_msgs_for_remote(
-            self._destination, last_device_stream_id, to_device_stream_id
+        last_device_list = self._last_device_list_stream_id
+        now_stream_id, results = yield self._store.get_devices_by_remote(
```

Comment:
It seems that get_devices_by_remote has a hard coded limit of 20 elements.
So we start with this request to be able to complete with the following one.

Reply:
makes sense. I think it would be helpful to include a comment that says `get_devices_by_remote` will return no more than 20 entries.